### PR TITLE
Add timeout to get remote config

### PR DESCRIPTION
**Pull request by atabel**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
     "javascript.validate.enable": false,
-    "eslint.validate": ["typescript"]
+    "eslint.validate": [
+        "typescript"
+    ],
+    "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (atabel) [inline .vscode/settings.json]: this forces vscode to use the TS version installed in node_modules

File: src/utils.ts [hunk 2]
```diff
@@ -87,10 +87,21 @@ const isRemoteConfigAvailable = (key: string) =>
 
 export const isABTestingAvailable = (key: string): Promise<boolean> => {
     if (!remoteConfig) {
-        return postMessageToNativeApp({type: 'GET_REMOTE_CONFIG'}).then(res => {
+        // If GET_REMOTE_CONFIG takes more than 5s to respond, resolve to false
+        const timeoutP = new Promise<boolean>(resolve => {
+            setTimeout(() => {
+                resolve(false);
+            }, 5000);
```
Comment (pladaria) [inline src/utils.ts]: To me this is too high, 500ms or less should be enough

File: src/utils.ts [hunk 3]
```diff
@@ -87,10 +87,21 @@ const isRemoteConfigAvailable = (key: string) =>
...
+        });
+
+        const configP = postMessageToNativeApp({
+            type: 'GET_REMOTE_CONFIG',
+        }).then(res => {
             remoteConfig = res;
             return isRemoteConfigAvailable(key);
         });
+
+        return Promise.race([timeoutP, configP]);
```
[the remark this answers is not in the document]
Comment (pladaria) [reply src/utils.ts]: hmm, I see that browser support for race is good

File: src/utils.ts [hunk 4]
```diff
@@ -87,10 +87,21 @@ const isRemoteConfigAvailable = (key: string) =>
 
 export const isABTestingAvailable = (key: string): Promise<boolean> => {
```
Comment (pladaria) [inline src/utils.ts]: Perhaps we should move this timeout logic to postMessageToNativeApp function

Comment (atabel) [reply src/utils.ts]: for the moment I'd include it only for remoteConfig

File: src/utils.ts (hunk 2 above)
Comment (atabel) [reply src/utils.ts]: 500ms can be too optimistic, in case this needs a network request, maybe 1 or 2 seconds is fine

Comment (pladaria) [reply src/utils.ts]: But this never waits for a network request it always returns the cached value or empty object

Comment (atabel) [reply src/utils.ts]: ok, set to 500

**Pull request merged**